core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -993,6 +999,18 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleApiVersionRequest(request: RequestChannel.Request) {
+    val responseHeader = new ResponseHeader(request.header.correlationId)
+    def isApiVersionRequestVersionSupported: Boolean = {
```

Comment:
@granthenke, it's a good point to mention that PR. Note, however, that we don't have error codes in every response type, so the behaviour will not be uniform, unfortunately (in some cases, the best we can do is disconnect).